### ISSUE TITLE
Infiltration Editor Meeting

### ISSUE THREAD

**Issue by FLO-2DKaren** [FLO-2DSoftware/qgis-flo-2d-plugin#956]
@rpachaly 
I figured out a great project that will engage your engineering skills and your python skills.  Let's do a meeting and see if we can start a plan.

**Comment by FLO-2DKaren** [comment]
@rpachaly 

OK I'll set it for 1pm and if you can't make it exactly at that time, just let me know.  The meeting can be reused or restarted at anytime.

Raincell and Infiltration Editor Meeting
Jun 14, 2023, 1:00 – 1:30 PM (America/New_York)

Please join my meeting from your computer, tablet or smartphone.
https://meet.goto.com/820619765
You can also dial in using your phone.
Access Code: 820-619-765
United States: [+1 (571) 317-3122](tel:+15713173122,,820619765)
- One-touch: tel:+15713173122,,820619765#


Get the app now and be ready when your first meeting starts:
https://meet.goto.com/install

**Comment by rpachaly** [comment]
1PM works for me!

**Comment by FLO-2DKaren** [comment]
[Soils-LC-AZ-Eloy.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/11749347/Soils-LC-AZ-Eloy.zip)

**Comment by FLO-2DKaren** [comment]
Load the infiltration data and build a small project to use it on.

**Comment by FLO-2DKaren** [comment]
More Ideas
**Soil**
Use Curve Number Generator like code to sample SURRGO for:
soil - chorizon
and any table that links the chorizon ID to the MUKey
[NDOT_Rainfall_Loss_Parameters JEFULLER.pdf](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/11770482/NDOT_Rainfall_Loss_Parameters.JEFULLER.pdf)


Use JE Fuller methods to calculate DTHETA, PSIF, and XKSAT.

**Landuse** 
Land use data is the main problem.  Curve number generator uses the rasters on NCLD.  That might be OK for a base resolution but it kind of sucks for roads, and paved impervious areas.  
Let's explore using OpenStreetMap for our impervious overlay.  It's good data for practically the whole world.

- Buildings
- Paved Surfaces
- Highways

**Comment by rpachaly** [comment]
Let me see if I understood this issue correctly for the **SOIL**:

1. SSURGO data: Download relevant soil data from NRCS SSURGO database - hzdept, hzdepb, sandtotal, silttotal, claytotal, fragsize, fragvol - from the area of interest (a.k.a computational domain) following the Curve Number Generator code. 
2. JE Fuller methods: Use the equations presented on the section 2.4 to calculate DTHETA, PSIF, and XKSAT for the downloaded data.
3. Use this calculated DTHETA, PSIF, and XKSAT to compute the Green-Ampt.

Basically, I need to add a new method on the Calculate Green-Ampt dialog that performs JE Fuller methods for estimating DTHETA, PSIF, and XKSAT instead of selecting directly the DTHETA, PSIF, and XKSAT from the soil layer. Is that correct?

**Comment by rpachaly** [comment]
**LAND USE:**

I was looking for anything that could help us to identify the roads, buildings, and paved surfaces and I've found this plugin [QuickOSM](https://github.com/3liz/QuickOSM/tree/master). Take a quick look in this [youtube video](https://www.youtube.com/watch?v=qQyyDxgObd8&ab_channel=OmarAlThuwaynee). Basically, this plugin can obtain the streets from OpenStreetMap. However, it does not return a raster but a shapefile. This is a problem, but I was checking and they have the number of lanes in the attribute table. Maybe we can apply a buffer based on the lanes and then convert it to a raster. After that, we can set these areas with a very high percent impervious area. What do you think? I've never seen this plugin before, but it looks promising.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
I was just looking at this a few days ago.  I think it is a good resource.

I wonder why the streets are coming down as polylines.  Because if you look at the OSM map, they are clearly defined as polygons.  I bet we can get them as polygons a different way.

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424460/79bb07ed-12e5-4cb6-8c0a-09150314d74b)

**Comment by FLO-2DKaren** [comment]
Hi Robson, 

That's kind of what I was thinking for the SSURGO or STATSGO data.  Put it into our own processor like the Curve Number Generator but for Green-Ampt data.

**Comment by rpachaly** [comment]
> @rpachaly I was just looking at this a few days ago. I think it is a good resource.
> 
> I wonder why the streets are coming down as polylines. Because if you look at the OSM map, they are clearly defined as polygons. I bet we can get them as polygons a different way.
> 
> ![image](https://user-images.githubusercontent.com/20424460/249484833-79bb07ed-12e5-4cb6-8c0a-09150314d74b.png)

Yeah, I think it could be possible to do that. 

> Hi Robson,
> 
> That's kind of what I was thinking for the SSURGO or STATSGO data. Put it into our own processor like the Curve Number Generator but for Green-Ampt data.

Awesome! I'll start working on this today. I'll add a button to the Compute Green-Ampt to run this new method for computing the Green-Ampt parameters. We can change the place of this button later, but the code for calculating everything will be working. 

Also, did you receive any information regarding the New Green-Ampt Method issue?

**Comment by rpachaly** [comment]
Just letting you know that I'm working on this and a lot of progress was made. 

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/a466b942-6b8f-415c-8068-fa119a40996c)

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/91c3c8a1-60f3-4ac0-947a-4fdb93cc71c2)

I'm able to download the soil data required for calculating DTHETA, PSIF, and XKSAT using the JE Fuller methods. However, the Curve Number Plugin it is able to download data using a Post Request and a Wfs Request. I don't know why, but my post request misses some features (take a look on the picture). I'll try the wfs in the next days to see if I can get better results. I believe when this issue is solved, calculating the G&A parameters would be easy.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
If you get stuck on some of that soil data, we can ask for help from JEFuller.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
For next week, we need to allow the plugin to read all words regardless of capitalization for wet, saturated, dry, or normal.

**Comment by rpachaly** [comment]
Ok! This will be the first thing that I'll do tomorrow.

**Comment by rpachaly** [comment]
> read all words regardless of capitalization for wet, saturated, dry, or normal.

@FLO-2DKaren 
Hi Karen, 

Just finished and pushed the branch. I also added an exception for when there is no wet, saturated, dry or normal on the saturation field. I ran some tests here and all good. Let me know if everything is working as expected. 

I'll continue working on the JEFuller method for infiltration.

**Comment by rpachaly** [comment]
I'm facing a hard time here. There are two problems here:
![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/876462ee-d9e7-4355-a571-10e0d190a1d0)

1. No data available: I think the solution here is to just set the Global Infiltration Parameters in areas where no data is available. Sometimes we do have the polygon but NULL data for the required fields. What do you think?
2. This is where I'm stuck right now. When I prepare the query for getting the data from SSURGO database, I need to join the chorizon, component, and mupolygon tables. This is returning multiple values of hzdept, hzdepb, sandtotal, silttotal, claytotal, fragsize, fragvol for the same mupolygonkey. What should I do in this case? Averaging out the parameters makes sense? I've been studying the SSURGO database, but I'm not sure about the best way to proceed with this.

**Comment by FLO-2DKaren** [comment]
1. No Data Available.  Generic Green-Ampt parameters don't always match.  The tables for HEC-RAS are different than the Green Ampt tables for AZDOT and NVDOT.  So let's ask JE Fuller and see what they say.
2. JE Fuller only did the top 3 inches of soil for their variables.  Make sure you are only looking at the codes for the top 3 inches.
3. If there are more than one value for the top 3 inches, we can average or we can take the confining value.  I think we want to limit the infiltration not increase it.  I think we want the more restrictive value.
4. JE Fuller might have used STATSGO for their data.  It seems to be soil data specific to states.  I would rather use more generic data since we hope to offer it as a method in other countries.

**Comment by rpachaly** [comment]
Some preliminary results and updates here:

[results.pptx](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/11952031/results.pptx)

In this presentation, I'm bringing the results comparing WPOINT, FCAPAC, SAT, PSIF, XKSAT, DTHETANrm, DTHETADry from the soil layer provided and the FE Fuller method. In my opinion, we're getting satisfactory results for WPOINT, FCAPAC, SAT, and PSIF. However, the XKSAT and the DTHETAs are still not good. They are deviating too much from the soil data. The code isn't totally ready, I need to organize some blocks, add some progress bars and check it again to see if the calculations are correct.  

What do you think about my testing procedure? It would be nice to have another soil layer containing all these fields to test in a different location.  

* I'm taking the first soil horizon to perform the calculations. Sometimes it is deeper than 3 inches. If there are multiple data for the same horizon, I'm taking the average of these data.
* The comparison was performed considering the same symbology. 
* I'm using data from SSURGO. 

When I get almost done, I would like to have a meeting to go through all the steps that I performed. Just to check carefully if everything is making sense and that the code is clear.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Hi Robson,

You are using the AZ DOT data.  It is a bit older than the NV report and it's DTHETA columns are mixed up.

Let's use the Nevada Data or the Maricopa County Data.  That way we know we're comparing the same process.

Maricopa County 
https://flo-2d.sharefile.com/d-sf1379a05d6b044a39486311e03ef4e26

Nevada 
[GreenAndAmpt_STATSGO2.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/11952526/GreenAndAmpt_STATSGO2.zip)

**Comment by rpachaly** [comment]
Hi Karen, 

I agree with you! Having a meeting and brainstorming is always good. I just started working here. When you are available, just sent me the link that I'll join the meeting.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Here is a link.

Green-Ampt Method
Jul 5, 2023, 12:30 – 1:00 PM (America/New_York)

Please join my meeting from your computer, tablet or smartphone.
https://meet.goto.com/137639309
You can also dial in using your phone.
Access Code: 137-639-309
United States: [+1 (224) 501-3412](tel:+12245013412,,137639309)
- One-touch: tel:+12245013412,,137639309#


Get the app now and be ready when your first meeting starts:
https://meet.goto.com/install

**Comment by FLO-2DKaren** [comment]
@rpachaly 

Notes from meeting.

Look at how curve number generator message that happens when the SSURGO site is down.
Curve number generator used to limit the size of the download layer.
We need to test this on our side.
CNG transforms the data to the project CRS.  Let's do that too.

Find out how JE Fuller decided which data to use when there is more than one.
Find out what they do when the first layer is less than 3 inches.

Testing
Run testing to compare to Nevada Shapefile.

Unit conversion
There should be a metric switch for one of the tables in the plugin.
Final units for English are inches or inches per hour
Final units for Metric are mm or mm/hr

Let's add module documentation for all the definitions

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Thanks Robson,  I'm excited for this new code.

**Comment by rpachaly** [comment]
You're welcome Karen! Thanks for the brainstorming.

**Comment by rpachaly** [comment]
Hi Karen, 

I didn't find any reference on how to calculate the soil's rock outcrop field. Do you have any material that shows how to calculate this field from the soil data? Or is there any other place where I can get this data?

**Comment by FLO-2DKaren** [comment]
I think it is in a table called map unit composition.  I haven't been able to access that table but I see it in the soil report.
Map Unit Composition
I got this database here that has a lot of rock outcrop.  Maybe  you can find it with an access sql query.

[wss_aoi_2023-07-07_14-55-08.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/11985709/wss_aoi_2023-07-07_14-55-08.zip)

**Comment by rpachaly** [comment]
Hi Karen, 

Some updates here. I've finished the portion of the code for the SSURGO data. I just need to run some tests here, but I already covered the great majority of the points that we discussed last week:

- Layers are converted to the project CRS

	I convert to WGS84 to create the post request on the NRCS database, then I convert it back again to the project’s CRS. 

- Units 

	Metric are switched based on the CRS project units.

- Holes 

	The holes were filled based on the closest feature. This was very complicated because the holes are not always inside a feature. In this case, the fill the holes processing algorithm works fine. However, sometimes they are in-between two or more features. That’s why I used the approach of selecting the closest feature. 

- Huge computational domain

	Bigger CDs crash the code due to the post request. Here is the solution that I’ve thought: Let’s break the computational domain in smaller ones, and then execute the code. I think this is not an easy task to perform, but creating this sort of algorithm for FLO 2D could be useful in other parts of the code. What do you think?

- Rock outcrop

	I still didn’t figure out a way of obtaining this data.

At this point I didn't find a good approach to use the STATSGO2 dataset to calculate the G&A parameters. I was reading, and as far as I understood, the STATSGO data is derived from SSURGO data. SSURGO provides detailed and local-level soil information, while STATSGO provides a more generalized view of soil properties at a regional scale. In my opinion, the SSURGO data is better for our purpose.

**Comment by rpachaly** [comment]
I also see the rockoutcrop on the table, but when I perform the query it returns an empty dictionary. I'll be focusing on this today. If the data is there, there is a way of getting it.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Add this to your search terms.
PERC_ROCK or Percent Rock

I also found this training manual from NDOT.

[Green_And_Ampt_Self_Help_Training.pdf](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12015091/Green_And_Ampt_Self_Help_Training.pdf)

**Comment by FLO-2DKaren** [comment]
I don't think the Curve Number Generator is working properly for Rock Outcrop.

Here is a test area.
![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424460/2cf5560f-0246-4b62-9507-365d7274ecc6)

The muname is RockLand
![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424460/deea1c81-6267-4156-a895-db77ba632e81)

The curve number is 62 which is not appropriate.  This area is like 95% rock coverage.

NLCD doesn't seem to recognize rock outcrop either.

**Comment by FLO-2DKaren** [comment]
I read this in the Self Help Training.

Soils imperviousness is included in the NDOT soils data shapefiles as PERC_ROCK or Percent Rock. This
terminology is consistent with the original NRCS soils data. For the purposes of consistency with Green and
Ampt reference documentation, Soils (Natural) Imperviousness is used in this document.

**Comment by rpachaly** [comment]
Yeah, I think that getting rockoutcrop data from NRCS is going to be a pain in the neck. But I'm getting the rockoutcrop data through the component table, just need to adjust some stuff in the code. Then we can test to see if the data worth it. After that I'll look into this percent rock.

**Comment by rpachaly** [comment]
The rockoutcrop data is from the STATSGO2 dataset. I cannot combine it with the SSURGO data.

**Comment by rpachaly** [comment]
> The rockoutcrop data is from the STATSGO2 dataset. I cannot combine it with the SSURGO data.

Forget this.

**Comment by rpachaly** [comment]
From the AOI that you sent me:

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/d8032083-bcda-4463-8441-b2b8c899d04b)

20% rock outcrop from the NRCS dataset.

What do you think?

**Comment by rpachaly** [comment]
> I don't think the Curve Number Generator is working properly for Rock Outcrop.
> 
> Here is a test area. ![image](https://user-images.githubusercontent.com/20424460/252683743-2cf5560f-0246-4b62-9507-365d7274ecc6.png)
> 
> The muname is RockLand ![image](https://user-images.githubusercontent.com/20424460/252684191-deea1c81-6267-4156-a895-db77ba632e81.png)
> 
> The curve number is 62 which is not appropriate. This area is like 95% rock coverage.
> 
> NLCD doesn't seem to recognize rock outcrop either.

I just ran the code for this location and no rock outcrop was found.

**Comment by FLO-2DKaren** [comment]
Haha Soooo Annoying. The polygon is literally named RockLand

**Comment by rpachaly** [comment]
Can we have a meeting if you're available?

**Comment by FLO-2DKaren** [comment]
@rpachaly 
I just opened this one.
QGIS Green Ampt Development Meeting
Tue, Jul 11, 2023 3:30 PM - 4:00 PM (EDT)

Please join my meeting from your computer, tablet or smartphone.
https://meet.goto.com/804029693

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Soil Data
Once you are satisfied with your process, test it in another two states.
Colorado
Alabama

Landuse Data
Assess the possibility of a base layer from OSM.
Initial saturation condition
   Dry, Normal, Wet
Vegetative Cover %
Initial Abstraction - mm or in

Impervious
Roadways
building
Paving

**Comment by rpachaly** [comment]
Hi Karen

I tested the methodology on Alabama and Colorado states. Here are the projects:

ALABAMA (meters)
[alabama_project.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12030060/alabama_project.zip)

COLORADO (feet)
[colorado_project.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12030127/colorado_project.zip)

It is interesting Colorado has rock outcrop data. I don't have a lot of experience in the G&A parameters to evaluate if they are in the correct range - that's why I'm sending the projects here. 

There is another thing. The SSURGO database has KSAT data. Their definition is: "The amount of water that would move vertically through a unit area of saturated soil in unit time under unit hydraulic gradient.". This is the saturated hydraulic conductivity. The JE Fuller method calculates the saturated hydraulic conductivity and then multiplies it by a factor of 0.5 to obtain the hydraulic conductivity. This is stated on the NDOT JE Fuller methods: 

"The Saxton and Rawls (2006) equations yield Ks, or saturated hydraulic conductivity, not effective hydraulic conductivity. Ks was multiplied by the correction factor of 0.5 to obtain Conductivity based on work of Bouwer (1966). This correction factor is consistent with the methods used in the 2014 ADOT Hydrology Manual."

If we are not satisfied with the hydraulic conductivity data, we can use SSURGO ksat * 0.5 and use the other parameters from the JE Fuller method (dthetas & PSIF).

**Comment by FLO-2DKaren** [comment]
@rpachaly 

I am also very new to developing the Green-Ampt parameters.  I trust the JE Fuller methodology.  They have been working on that method for over a decade.  

We can apply whatever methods JE Fuller use for xksat, psif, and dtheta.

When I review your data, I will use generic lookup table to spot check a range of values.  The main thing I'm hoping for is general estimates being in a good range and the max and min values to be reasonable.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Notes from my initial inspection.
I couldn't download the Alabama data.  Maybe send it to me directly.

1. Add a Floor value so that xksat doesn't go below 0.01 in/hr or 2.54 mm/hr

XKSAT(natural) = XKSAT(full saturation) * KsCF
If XKSAT(natural) < 0.01 Then
 **XKSAT(natural) = 0.01**

2. Add a Soil Description for the final soil layer.  
![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424460/4268b6d0-58f2-4b3c-8fe9-2bad492719af)

3. XKSAT numbers seem low.  I see 0.33 to 0.5 for sandy loam in the Maricopa data.   Our data is less than 0.08 for sandy loam.

**Comment by rpachaly** [comment]
[alabama_project.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12030893/alabama_project.zip)

This one should work.

**Comment by rpachaly** [comment]
I sent the Alabama project to your email. The Maricopa data isn't correct. I'm going to send you the correct data.

**Comment by FLO-2DKaren** [comment]
@rpachaly 

Thanks, I got the Alabama Data.

I'm referring to this Maricopa data.  I'm not using the other data.  This is the Brand New data from this year.
![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/20424460/932e54bc-49d1-492d-991b-eba091a4971f)

https://flo-2d.sharefile.com/d-s843f67b0b5644341aa5a0ad258d295db

**Comment by rpachaly** [comment]
Yeah, I've a project with this data. In this project, the CRS units are in feet (in/hr) and the GAParametersFinal_June2023_rep XKSAT2006 are in mm/hr. I converted the data and on the soil_layer we have hydc and hydc_si.

[maricopa.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12031108/maricopa.zip)

I also ran some statistics here:

soil_layer
{'CV': 0.6680307651895053,
'FIRSTQUARTILE': 0.076,
'IQR': 0.178,
'MAJORITY': 0.254,
'MAX': 0.762,
'MEAN': 0.156027397260274,
'MEDIAN': 0.152,
'MIN': 0.025,
'MINORITY': 0.762,
'RANGE': 0.737,
'STD_DEV': 0.10423110158230778,
'THIRDQUARTILE': 0.254}

GAParametersFinal_June2023_rep
{'CV': 0.9450196695093711,
'FIRSTQUARTILE': 0.049,
'IQR': 0.199,
'MAJORITY': 0.248,
'MAX': 0.99,
'MEAN': 0.14055045871559624,
'MEDIAN': 0.093,
'MIN': 0.022,
'MINORITY': 0.156,
'RANGE': 0.968,
'STD_DEV': 0.13282294804480327,
'THIRDQUARTILE': 0.248}

**Comment by rpachaly** [comment]
@FLO-2DKaren 
I'm facing a tropical thunderstorm here and my connection started to get really unstable. I'll have to call it a day. Tomorrow I'll be back working on this.

**Comment by FLO-2DKaren** [comment]
@rpachaly OK sounds good.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
When you get back.  I'm looking in the Chapter 4 document and it is also showing Sandy Loam to be closer to 0.3 to 0.5.

**Comment by FLO-2DKaren** [comment]
I don't think that data is in mm/hr.  It should also be in in/hr.

**Comment by rpachaly** [comment]
Hi Karen,

You were right, the data is in in/hr. I was checking the code and there are some import points to be made. My code now follows exactly the spreadsheet available at https://hrsl.ba.ars.usda.gov/SPAW/Index.htm, which is the spreadsheet used to check the computations made using the NDOT manual. The code and the spreadsheet results are matching. Here is the spreadsheet: 
[Soil Water Characteristics-Equations.xls](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12040671/Soil.Water.Characteristics-Equations.xls)


1.	On the NDOT manual, when the XKSAT is calculated, there is a constant (0.0393700787) being applied to the calculating that is not present in the spreadsheet. This is probably a unit conversion. I removed because I my calculations are performed in SI units and then transformed to the user's unit.
2.	The min and maximum values on the NDOT manual are in in/hr. My code was setting these values in mm/hr. Now it is fixed. XKSAT min is 0.254 mm/hr (0.01 in/hr) or 50.8 mm/hr (2 in/hr).
3.	The muname is added to the soil_layer.

I used the maricopa data to check the results. This is the project that I'm using: [maricopa_2.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/12040693/maricopa_2.zip). The results are very close now and, in most features, they are matching. 

If you think that we are achieving good results, I can push the branch.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Yes for sure push the branch so I can see your calculations too.  

This is great news.

**Comment by rpachaly** [comment]
I pushed the branch!

**Comment by FLO-2DKaren** [comment]
@rpachaly 

What if we don't need to apply  standard processors because we use the raster data instead of polygon data for the land use. 
 This complies with the last option pretty well.  We don't need that data to be in polygon format.  It still means processing but I think we could do it with zonal statistics or something like that.

**Comment by rpachaly** [comment]
That's a good approach. It will make everything faster.

**Comment by rpachaly** [comment]
@FLO-2DKaren 
Hi Karen, 

I did everything by hand today to see if it is doable. Here are the results:

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/7e02f8f1-deae-4f32-b133-0f0ee32c42b9)

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/bf82b215-4922-4fd1-9f55-5175219655ea)

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/33d89c81-3822-4b02-90f5-8898945b12bc)

I liked the results. It is a waaaay better than using the NLCD for urban areas. We can select everything that we want from the OSM based on the color. Then we can use the NDOT table to estimate IA, VC, and RTIMP. 

![image](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/assets/39889306/5e499687-5d71-4c33-8212-513808b8233c)
 
I'll start coding.

**Comment by FLO-2DKaren** [comment]
Yes that looks really good.

**Comment by rpachaly** [comment]
@FLO-2DKaren 
Hi Karen, 

I'm finishing up the land use tool, just need to run some tests here. Everything is looking really good. Can we have a meeting this afternoon?

**Comment by rpachaly** [comment]
That works for me!

Awesome! Are you using compression on the hdf5 file?

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Here is a meeting link.
I'm not using compression.  Noemi used compression on her hdf5 files and I was able to test them successfully. 

GreenAmpt Meeting
Jul 19, 2023, 12:00 – 12:30 PM (America/New_York)

Please join my meeting from your computer, tablet or smartphone.
https://meet.goto.com/311587053
You can also dial in using your phone.
Access Code: 311-587-053
United States: [+1 (408) 650-3123](tel:+14086503123,,311587053)
- One-touch: tel:+14086503123,,311587053#

Get the app now and be ready when your first meeting starts:
https://meet.goto.com/install

**Comment by FLO-2DKaren** [comment]
@rpachaly 
Robson,  Don't join the GoToMeeting.

Use this one
To join the video meeting, click this link: https://meet.google.com/sqh-dsur-evv
Otherwise, to join by phone, dial +1 413-752-0156 and enter this PIN: 999 769 115#
To view more phone numbers, click this link: https://tel.meet/sqh-dsur-evv?hs=5

**Comment by rpachaly** [comment]
New infiltration editor ready!

- SSURGO data for estimating soil G&A parameters
- OSM data for estimating land use G&A parameters
- Documentation added for the new infiltration editor
- Tested on cities in Arizona, Nevada, and Colorado
- Minor bugs fixed

I pushed the branch.

**Comment by FLO-2DKaren** [comment]
@rpachaly 
I don't know what the rocket means.  I assume it means launch.  
I merged this branch into the regular plugin so I can give it to a couple of people to test.

I'll update the meta data too.
